assets/src/dashboard/app/views/myStories/content/storyListItem/index.js

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+import { getRelativeDisplayDate } from '@web-stories-wp/date';
+import { __, sprintf } from '@web-stories-wp/i18n';
+import {
+  Headline,
+  Text,
+  THEME_CONSTANTS,
+  Tooltip,
+} from '@web-stories-wp/design-system';
+
+/**
+ * Internal dependencies
+ */
+import {
+  InlineInputForm,
+  StoryMenu,
+  TableCell,
+  TablePreviewCell,
+  TableStatusCell,
+} from '../../../../../components';
+import { generateStoryMenu } from '../../../../../components/popoverMenu/story-menu-generator';
+import { STORY_STATUS } from '../../../../../constants';
+import {
+  RenameStoryPropType,
+  StoryMenuPropType,
+  StoryPropType,
+} from '../../../../../types';
+import { titleFormatted } from '../../../../../utils';
+
+import {
+  LockIcon,
+  PreviewWrapper,
+  PreviewImage,
+  StyledTableRow,
+  TitleTableCellContainer,
+} from './components';
+
+function onFocusSelectAll(e) {
+  window.getSelection().selectAllChildren(e.target);
+}
+
+function onBlurDeselectAll() {
+  window.getSelection().removeAllRanges();
+}
+
+const DISPLAY_STATUS = {
+  [STORY_STATUS.PUBLISH]: __('Published', 'web-stories'),
+  [STORY_STATUS.FUTURE]: __('Scheduled', 'web-stories'),
+  [STORY_STATUS.DRAFT]: __('Draft', 'web-stories'),
+  [STORY_STATUS.PRIVATE]: __('Private', 'web-stories'),
+};
+
+export const StoryListItem = ({
+  story,
+  renameStory,
+  userId,
+  storyMenu,
+  storyStatus,
+}) => {
+  const isLocked = useMemo(
+    () => story?.locked && userId !== story?.lockUser.id,
+    [story, userId]
+  );
+
+  const formattedTitle = titleFormatted(story.title);
+
+  const memoizedStoryMenu = useMemo(
+    () => (
+      <StoryMenu
+        menuLabel={
+          isLocked
+            ? sprintf(
+                /* translators: %1$s: story title %2$s: user story is locked by.*/
+                __('Context menu for %1$s (locked by %2$s)', 'web-stories'),
+                formattedTitle,
+                story?.lockUser.name
+              )
+            : sprintf(
+                /* translators: %s: story title.*/
+                __('Context menu for %s', 'web-stories'),
+                formattedTitle
+              )
```

Comment:
Does it make sense to change the label depending on the lock?

Whether the story is locked is already announced by the `Headline` below.

So this label could just describe the menu itself as otherwise it's a bit redundant.

Reply:
great point, yeah it's redundant. i forgot that in the list view the story titles can be focused with the keyboard while the grid view the context menu now acts as the general label for the story.